src/heartbeat/heartbeat.c

```diff
@@ -78,6 +81,21 @@ volatile bool hb_send_obc_resp = false;
 volatile bool hb_send_eps_resp = false;
 volatile bool hb_send_pay_resp = false;
 
+// Current SSM -> Request reset data from other SSM
+volatile bool hb_send_obc_rdata_req = false;
+volatile bool hb_send_eps_rdata_req = false;
+volatile bool hb_send_pay_rdata_req = false;
+
+// True if we received reset data response
+volatile bool hb_received_obc_rdata_resp = false;
+volatile bool hb_received_eps_rdata_resp = false;
+volatile bool hb_received_pay_rdata_resp = false;
+
+// Other SSM -> Send restart data from current SSM
+volatile bool hb_send_obc_rdata_resp = false;
+volatile bool hb_send_eps_rdata_resp = false;
+volatile bool hb_send_pay_rdata_resp = false;
+
```

Comment:
I just realized you need a way for the device requesting restart info to return that obtained information back to the calling code where you initiate the request (e.g. in the diagnostic, how do you retrieve the requested data to figure out if it worked?). You should think about how to do that and we can discuss it.

Reply:
Currently, we have the restart info sent as part of the tx callback in heartbeat.c for opcodes 3 and 4. I think that adding functionality to print the appropriate fields of the tx callbacks would be sufficient to obtain this information.